onnxruntime/core/session/inference_session.cc

```diff
@@ -3486,6 +3486,10 @@ common::Status InferenceSession::Run(IOBinding& io_binding) {
 
 template <typename T>
 void InferenceSession::StartProfiling(const std::basic_string<T>& file_prefix) {
+  if (session_profiler_.IsEnabled()) {
+    LOGS(*session_logger_, WARNING) << "Profiler is already running.";
+    return;
+  }
```

Comment:
Do you consider the situation that users call session.StartProfling multiple times without calling session.EndProfiling, like below, will that cause any errors or crashes?
```
sess.StartProfiling()
xxx
xxx
sess.StartProfiling()
xxx
xxx
sess.EndProfiling()
```